[Docs] Add documentation about storage indexers (#1374)
elastic/package-registry: CHANGELOG.md

### Added

> * Set storage indexer feature, previously in technical preview, is now GA. [#1374](https://github.com/elastic/package-registry/pull/1374)

Author: Do you think this should be rephrased?

Reviewer: ```suggestion
* Storage indexer based on Google Cloud Storage is now GA. [#1374](https://github.com/elastic/package-registry/pull/1374)
```

* Added cache for categories endpoint (SQL storage indexer). [#1379](https://github.com/elastic/package-registry/1379)
* Update default values for cache search and TTL settings search (SQL storage indexer). [#1379](https://github.com/elastic/package-registry/1379)

> 

Reviewer: Nit. Empty line?
```suggestion
```
